trieste/bayesian_optimizer.py

```diff
@@ -204,12 +231,23 @@ def try_get_final_model(self) -> TrainableProbabilisticModel:
     @property
     def loaded_history(self) -> list[Record[StateType]]:
         """The history of the optimization process loaded into memory."""
-        return [record.load() for record in self.history]
+        return [record if isinstance(record, Record) else record.load() for record in self.history]
+
+    def save_result(self, path: Path) -> None:
```

Comment:
if files with same names are present this will overwrite them, right? we should probably mention in it in the docs